src/BaseControllerV2.ts

```diff
@@ -120,3 +170,56 @@ export class BaseController<S extends Record<string, unknown>> {
     this.internalListeners.clear();
   }
 }
+
+// This function acts as a type guard. Using a `typeof` conditional didn't seem to work.
+function isStateDerivingFunction<T>(x: boolean | DerivedState<T>): x is DerivedState<T> {
+  return typeof x === 'function';
+}
+
+/**
+ * Returns an anonymized representation of the controller state.
+ *
+ * By "anonymized" we mean that it should not contain any information that could be personally
+ * identifiable.
+ *
+ * @param state - The controller state
+ * @param metadata - The controller state metadata, which describes how to derive the
+ *   anonymized state
+ * @returns The anonymized controller state
+ */
+export function getAnonymizedState<S extends Record<string, any>>(
+  state: S,
+  metadata: StateMetadata<S>,
+): Record<string, unknown> {
+  return deriveStateFromMetadata(state, metadata, 'anonymous');
+}
+
+/**
+ * Returns the subset of state that should be persisted
+ *
+ * @param state - The controller state
+ * @param metadata - The controller state metadata, which describes which pieces of state should be persisted
+ * @returns The subset of controller state that should be persisted
+ */
+export function getPersistentState<S extends Record<string, any>>(
+  state: S,
+  metadata: StateMetadata<S>,
+): Record<string, unknown> {
+  return deriveStateFromMetadata(state, metadata, 'persist');
+}
+
+function deriveStateFromMetadata<S extends Record<string, any>>(
+  state: S,
+  metadata: StateMetadata<S>,
+  metadataProperty: 'anonymous' | 'persist',
+): Record<string, unknown> {
```

Comment:
... just realized something. This return type makes `IsJsonable<>` impossible to apply because of the `unknown` :grimacing: 

Back to the drawing board I guess!

Reply:
Ok, I think I've got it this time! I created another JSON type - one without a generic type parameter.